src/main/java/io/sniffy/Sniffer.java

```diff
@@ -55,11 +57,24 @@ private static synchronized void notifyListeners(StatementMetaData statementMeta
         }
     }
 
+    private static synchronized void notifyListeners(InetSocketAddress address, int connectionId, SocketStats socketStats) {
+        Iterator<WeakReference<Spy>> iterator = registeredSpies.iterator();
+        while (iterator.hasNext()) {
+            WeakReference<Spy> spyReference = iterator.next();
+            Spy spy = spyReference.get();
+            if (null == spy) {
+                iterator.remove();
+            } else {
+                spy.addSocketOperation(address, new SocketStats(socketStats));
+            }
+        }
+    }
+
     // query counters
 
-    protected static final Counter COUNTER = new Counter();
+    static final Counter COUNTER = new Counter();
 
-    protected static final ThreadLocal<Counter> THREAD_LOCAL_COUNTER = new ThreadLocal<Counter>() {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://www.codacy.com/app/sniffy/sniffy/file/2492520258/issues/source?bid=3005960&fileBranchId=3034483#l77)